scopesim/effects/effects.py

```diff
@@ -105,9 +108,193 @@ def include(self):
+    
+    
+        """
+        changes = self.meta.get("changes", [])
+        changes_str = "- " + "\n- ".join([str(entry) for entry in changes])
+        cls_doc = self.__doc__ if self.__doc__ is not None else "<no docstring>"
+        cls_descr = cls_doc.lstrip().splitlines()[0]
+
+        params = {"report_plot_filename": None,
+                  "report_plot_file_formats": ["png"],
+                  "report_plot_caption": "",
+                  "report_plot_include": False,
+                  "report_table_caption": "",
+                  "report_table_include": False,
+                  "report_image_path": "!SIM.reports.image_path",
+                  "report_rst_path": "!SIM.reports.rst_path",
+                  "report_latex_path": "!SIM.reports.latex_path",
+                  "file_description": self.meta.get("description",
+                                                    "<no description>"),
+                  "class_description": cls_descr,
+                  "changes_str": changes_str}
+        params.update(self.meta)
+        params.update(kwargs)
+        params = from_currsys(params)
+
+        rst_str = """
+{}
+{}
+**Included by default**: ``{}``
+
+**File Description**: {}
+
+**Class Description**: {}
+
+**Changes**:
 
+{}
+
+Data
+{}
+""".format(str(self),
+           rst_title_chars[0] * len(str(self)),
+           params["include"],
+           params["file_description"],
+           params["class_description"],
+           params["changes_str"],
+           rst_title_chars[1] * 4)
+
+        if params["report_plot_include"] and hasattr(self, "plot"):
+            fig = self.plot()
+
+            path = params["report_image_path"]
+
+            fname = params["report_plot_filename"]
+            if fname is None:
+                fname = self.meta["name"].lower().replace(" ", "_")
+
+            for fmt in params["report_plot_file_formats"]:
+                fname = ".".join((fname.split(".")[0], fmt))
+                file_path = os.path.join(path, fname)
+
+                fig.savefig(fname=file_path)
+
+            # rel_path = os.path.relpath(params["report_image_path"],
+            #                            params["report_rst_path"])
+            # rel_file_path = os.path.join(rel_path, fname)
+
+            rst_str += """
+.. figure:: {}
+    :name: {}
+
+    {}
+""".format(fname,
+           "fig:" + params.get("name", "<unknown Effect>"),
+           params["report_plot_caption"])
+    
+        if params["report_table_include"]:
+            rst_str += """
+.. table::
+    :name: {}
+
+{}
+
+{}
+""".format("tbl:" + params.get("name"),
+           table_to_rst(self.table, indent=4),
+           params["report_table_caption"])
+
+        rst_str += """
+Meta-data
+{}
+::
+
+{}
+""".format(rst_title_chars[1] * 9,
+           self.meta_string)
+    
+        write_report(rst_str, filename, output)
+
+        return rst_str
+
+    def __repr__(self):
```

Comment:
`__repr__` should be something that when you type it in you get the same object I think (where that is feasible)

Reply:
I need to look into the proper usage of __repr__ and __str__. But thanks for pointing it out